### PR TITLE
feat(parser): parse `CALLED ON NULL INPUT` as a property instead of `Command`

### LINKED ISSUE #4916: sqlglot parsing issue with Snowflake's `CALLED ON NULL INPUT` clause
## Issue Description
When parsing Snowflake UDFs/PROCEDUREs that includes the `CALLED ON NULL INPUT` clause in user-defined function and stored procedure definitions, sqlglot incorrectly creates a `Command` expression instead of a proper `Create` expression. This causes downstream issues when trying to analyze or manipulate the AST.

## Reproducible Code Snippet

```python
import sqlglot

# This SQL should parse into a Create expression but results in a Command expression
problem_sql = """
CREATE OR REPLACE FUNCTION func.example_function(x FLOAT)
RETURNS FLOAT
LANGUAGE SQL
CALLED ON NULL INPUT
AS
$$
  RETURN x * 2;
$$;
"""

# Parse with correct dialect specified
ast = sqlglot.parse_one(problem_sql, read="snowflake")
print(f"Expression type: {type(ast)}")  # Shows sqlglot.expressions.Command instead of Create

# This version works correctly without the CALLED ON NULL INPUT clause
working_sql = """
CREATE OR REPLACE FUNCTION func.example_function_working(x FLOAT)
RETURNS FLOAT
LANGUAGE SQL
AS
$$
  RETURN x * 2;
$$;
"""

ast_working = sqlglot.parse_one(working_sql, read="snowflake")
print(f"Working version expression type: {type(ast_working)}")  # Shows sqlglot.expressions.Create
```

## Demonstrating the difference

```python
# Further inspect the difference in parsing
problem_ast = sqlglot.parse_one(problem_sql, read="snowflake")
working_ast = sqlglot.parse_one(working_sql, read="snowflake")

print("Problem SQL parsing:")
print(f"- Expression type: {type(problem_ast)}")
print(f"- Expression structure: {problem_ast}")

print("\nWorking SQL parsing:")
print(f"- Expression type: {type(working_ast)}")
print(f"- Expression structure: {working_ast}")
```

## Official Documentation

- [Snowflake CREATE FUNCTION documentation](https://docs.snowflake.com/en/sql-reference/sql/create-function#syntax) shows `CALLED ON NULL INPUT` as a valid option:
  ```
  CREATE [ OR REPLACE ] FUNCTION [ IF NOT EXISTS ] <name> ( [ <arg_name> <arg_data_type> ] [ , ... ] )
    RETURNS <result_data_type>
    [ { CALLED ON NULL INPUT | RETURNS NULL ON NULL INPUT | STRICT } ]
    ...
  ```

- [Snowflake CREATE PROCEDURE documentation](https://docs.snowflake.com/en/sql-reference/sql/create-procedure#syntax) also includes this clause:
  ```
  CREATE [ OR REPLACE ] PROCEDURE [ IF NOT EXISTS ] <name> ( [ <arg_name> <arg_data_type> ] [ , ... ] )
    RETURNS <result_data_type>
    [ { CALLED ON NULL INPUT | RETURNS NULL ON NULL INPUT | STRICT } ]
    ...
  ```

## Expected Behavior
sqlglot should parse the SQL with `CALLED ON NULL INPUT` into a proper `Create` expression, recognizing this clause as a valid part of the function/procedure definition in Snowflake SQL. The `CALLED ON NULL INPUT` should be captured as a property of the `Create` expression, not cause the parser to produce a `Command` expression.

## Environment Information
- sqlglot version: 26.11.1
- Python version: 3.11.6
- Operating system: MacOS 15

### PR DESCRIPTION
**Pull request by eruditmorina**
## Summary

Closes https://github.com/tobymao/sqlglot/issues/4916

`CREATE FUNCTION ... CALLED ON NULL INPUT` falls back to `exp.Command` for every dialect, losing the AST representation. This PR aims to fix it by adding `CalledOnNullInputProperty` so the statement parses correctly as `exp.Create`.

The parser hook is added to the `base` parser rather than scoped to a single dialect (original issue reports Snowflake). This fixes Snowflake, Postgres, BigQuery, and the base `sqlglot` dialect. Added tests to cover Snowflake, Postgres and extended `identity.sql` to validate the default `sqlglot` dialect.

### CODE REVIEW

File: tests/dialects/test_snowflake.py [hunk 1]
```diff
@@ -4427,6 +4427,10 @@ def test_user_defined_functions(self):
                 "snowflake": "CREATE FUNCTION a() RETURNS INT IMMUTABLE AS 'SELECT 1'",
             },
         )
+        self.validate_identity(
+            "CREATE FUNCTION a(x FLOAT) RETURNS FLOAT LANGUAGE SQL CALLED ON NULL INPUT AS $$ RETURN x * 2; $$",
```
Comment (geooo109) [inline tests/dialects/test_snowflake.py]: Did you run this in SF ? The `RETURN` inside the `$$ $$` shouldn't exist right ? Let's remove the `RETURN` to make it runnable.

File: sqlglot/parser.py [hunk 2]
```diff
@@ -2900,6 +2900,14 @@ def _parse_settings_property(self) -> exp.SettingsProperty:
             exp.SettingsProperty(expressions=self._parse_csv(self._parse_assignment))
         )
 
+    def _parse_called_on_null_input_property(self) -> exp.CalledOnNullInputProperty | None:
+        index = self._index
+        if self._match_text_seq("ON", "NULL", "INPUT"):
+            return self.expression(exp.CalledOnNullInputProperty())
+
+        self._retreat(index - 1)
+        return None
```
Comment (geooo109) [inline sqlglot/parser.py]:
```suggestion
    def _parse_called_on_null_input_property(self) -> exp.CalledOnNullInputProperty | None:
        if not self._match_text_seq("ON", "NULL", "INPUT"):
            self._retreat(self._index - 1)
            return None

        return self.expression(exp.CalledOnNullInputProperty())
```

**Review by geooo109 — COMMENTED** [review]
Nice work! left some comments to take into account

File: tests/dialects/test_snowflake.py (hunk 1 above)
Comment (eruditmorina) [reply tests/dialects/test_snowflake.py]: That's right, adjusted!

File: tests/dialects/test_snowflake.py [hunk 3]
```diff
@@ -4427,6 +4427,14 @@ def test_user_defined_functions(self):
...
             },
         )
+        self.validate_identity(
+            "CREATE FUNCTION a(x FLOAT) RETURNS FLOAT LANGUAGE SQL CALLED ON NULL INPUT AS $$ x * 2 $$",
+            "CREATE FUNCTION a(x DOUBLE) RETURNS DOUBLE LANGUAGE SQL CALLED ON NULL INPUT AS ' x * 2 '",
+        ).assert_is(exp.Create)
+        self.validate_identity(
+            "CREATE FUNCTION a(x FLOAT) RETURNS FLOAT LANGUAGE SQL RETURNS NULL ON NULL INPUT AS $$ x * 2 $$",
+            "CREATE FUNCTION a(x DOUBLE) RETURNS DOUBLE LANGUAGE SQL RETURNS NULL ON NULL INPUT AS ' x * 2 '",
+        ).assert_is(exp.Create)
```
Comment (geooo109) [inline tests/dialects/test_snowflake.py:4432]:
```suggestion
        self.validate_identity(
            "CREATE FUNCTION a(x DOUBLE) RETURNS DOUBLE LANGUAGE SQL RETURNS NULL ON NULL INPUT AS ' x * 2 '",
        ).assert_is(exp.Create)
```

Comment (eruditmorina) [reply tests/dialects/test_snowflake.py:4432]: question: shouldn't we keep the test that covers `CALLED ON NULL INPUT` here since that's the main test for the introduced changes? The one covering `RETURNS NULL ON NULL INPUT` might be duplicate here.

**Review by geooo109 — APPROVED** [review]

File: tests/dialects/test_snowflake.py (hunk 3 above)
Comment (geooo109) [reply tests/dialects/test_snowflake.py:4432]: @eruditmorina ugh, yeah sorry got confused.

```
self.valida_identity("CREATE FUNCTION a(x DOUBLE) RETURNS DOUBLE LANGUAGE SQL CALLED ON NULL INPUT AS ' x * 2 '").assert_is(exp.Create)
```

Let's use this ^, and remove the `RETURNS` one.

Comment (eruditmorina) [reply tests/dialects/test_snowflake.py:4432]: Done, thank you.

**Review by geooo109 — APPROVED** [review]
Great work, thank you!
